### ISSUE TITLE
Getting Multiple Webhooks

### ISSUE THREAD

**Issue by maddox** [jishi/node-sonos-http-api#490]
For some reason I'm getting multiple web hooks after a simple `join` action for a speaker.

You can see them in the log from my project:

```
Sonos: Hook triggered
Sonos Hook: zone state change
127.0.0.1 - - [20/Apr/2017 03:12:17] "POST /sonos/hook HTTP/1.1" 200 2 0.0007
Sonos: Hook triggered
Sonos Hook: topology change
127.0.0.1 - - [20/Apr/2017 03:12:18] "POST /sonos/hook HTTP/1.1" 200 2 0.0452
Sonos: Hook triggered
Sonos Hook: topology change
127.0.0.1 - - [20/Apr/2017 03:12:18] "POST /sonos/hook HTTP/1.1" 200 2 0.0852
Sonos: Hook triggered
Sonos Hook: topology change
127.0.0.1 - - [20/Apr/2017 03:12:18] "POST /sonos/hook HTTP/1.1" 200 2 0.0428
Sonos: Hook triggered
Sonos Hook: topology change
127.0.0.1 - - [20/Apr/2017 03:12:18] "POST /sonos/hook HTTP/1.1" 200 2 0.0397
Sonos: Hook triggered
Sonos Hook: topology change
127.0.0.1 - - [20/Apr/2017 03:12:18] "POST /sonos/hook HTTP/1.1" 200 2 0.0861
Sonos: Hook triggered
Sonos Hook: topology change
127.0.0.1 - - [20/Apr/2017 03:12:18] "POST /sonos/hook HTTP/1.1" 200 2 0.0425
Sonos: Hook triggered
Sonos Hook: topology change
127.0.0.1 - - [20/Apr/2017 03:12:18] "POST /sonos/hook HTTP/1.1" 200 2 0.0401
```

Is there any reason this might be happening that I can control?

**Comment by jishi** [comment]
I know that the topology changes are a bit "spammy" from Sonos itself, but that seems a bit excessive. Is this always the case? Even on a freshly restarted http-api? 

Is there any difference how big the group is?

**Comment by maddox** [comment]
it seems variable :/

When dev'ing a feature of my dashboard, it was fine. It was a single hook. But in production it was 2. Then I restarted and it was A TON.

It just sort of stinks because they don't seem to be eventually consistent.

**Comment by jishi** [comment]
Hm. Thinking about it, I don't do an UNSUBSCRIBE from events when exiting, so repeatedly exiting and restarting the api will subscribe multiple times to topology events. This should settle down after 18 minutes or so, since that is the subscription timeout (1100 seconds).

Could that be the explanation for the behavior you are seeing?

**Comment by maddox** [comment]
I bet it totally is! Since I was changing stuff rapidly and restarting, that would explain why I'd get multiple notifications one place and a single one elsewhere. And why the count would vary.

It'd def be worth doing an unsubscribe when exiting, just to have it clean up after itself.

**Comment by jishi** [comment]
Yeah, the problem is that the behavior is different on Linux vs Windows. On Windows you don't seem to get a SIGINT on a Ctrl + C so there is no way of handling it gracefully. Would work for Linux and MacOS though.

**Comment by maddox** [comment]
> Would work for Linux and MacOS though.

Works for me! lol.

**Comment by jishi** [comment]
I'll see what we can do. Actually, I've already implemented the unsubscription part (as a dispose() method) but it's not triggered by anything. https://github.com/jishi/node-sonos-discovery/blob/master/lib/Subscriber.js
